unresolved external symbol "unsigned char __cdecl sbepp::detail::byteswap<unsigned char>(unsigned char)"

When I build my application based on sbepp v1.3.0 with Visual Studio 2019/2022, I get a link time error:

```
error LNK2001: unresolved external symbol "unsigned char __cdecl sbepp::detail::byteswap<unsigned char>(unsigned char)" (??$byteswap@E@detail@sbepp@@YAEE@Z)
```

Looks like the problem is caused by the following declaration without the definition:

`sbepp\sbepp.hpp:(464)`
```cpp

// because `if((E == endian::native) || (sizeof(T) == 1))` is not a constexpr-if
// this function has to be declared for single byte types even if it will never
// be used
template<typename T>
T byteswap(T) noexcept;
```

g++ 11/12 compiles the code just fine.

Please let me know if you need a full source code to reproduce the issue.

--- a/sbepp/src/sbepp/sbepp.hpp
+++ b/sbepp/src/sbepp/sbepp.hpp
@@ -458,11 +458,14 @@ using std::byteswap;
 
 #else
 
-// because `if((E == endian::native) || (sizeof(T) == 1))` is not a constexpr-if
-// this function has to be declared for single byte types even if it will never
-// be used
-template<typename T>
-T byteswap(T) noexcept;
+// because `if((E == endian::native) || (sizeof(T) == 1))` used in
+// `get/set_primitive` is not a constexpr-if, this function has to be declared
+// (and defined for MSVC) for single byte types even if it will never be used
+template<typename T, typename = enable_if_t<sizeof(T) == 1>>
+constexpr T byteswap(T value) noexcept
+{
+    return value;
+}
 
 #    if defined(_MSC_VER) && (!defined(__clang__) || defined(__c2__))
 

